### DISCUSSION TITLE
filterset_class does not apply to patch action

### DISCUSSION THREAD

**Discussion by LeonardoFurtado** [carltongibson/django-filter#1519] [General]
I have a patch action like this:

```python
@action(detail=False, methods=['patch'])
    def do_something(self, request):
        foo = request.data.get('foo')
        if foo not is None:
            self.get_queryset().update(tipo_id=foo)
            return Response(status=status.HTTP_200_OK)
        return Response(status=status.HTTP_400_BAD_REQUEST)
```

my `request.data` also have an attribute named `bar` but my  `self.get_queryset().update(tipo_id=foo)` is not filtering by bar. I've tried to change de method of this action to get (`methods=['get']`) and the filter have worked fine.
Is this a bug ?

**Comment by shipengtaov** [comment]
According to django-rest-framework doc: <https://www.django-rest-framework.org/api-guide/filtering/>, the filter is only for list views. But you can still override `.get_queryset()` to change this behavior.
Or call `filter_queryset` in your action method like this:
<https://github.com/encode/django-rest-framework/blob/3.13.1/rest_framework/mixins.py#L38>
